server/src/main/java/org/elasticsearch/common/time/DateFormatter.java

```diff
@@ -30,6 +31,14 @@
      */
     TemporalAccessor parse(String input);
 
+    /**
+     * Try to parse input to a java time TemporalAccessor
+     * @param input An arbitrary string resembling the string representation of a date or time
+     * @return      Tuple containing a boolean value indicating parsing success or failure and a java time object containing
+     * the parsed input in the case of successful parsing
+     */
+    Optional<TemporalAccessor> parseWithoutException(String input);
```

Comment:
I do not think we should add another parse method. If this is more performant, then let's change the implementation of parse.